### DISCUSSION TITLE
Webcam no longer works

### DISCUSSION THREAD

**Discussion by mattcalavera** [Guilouz/Klipper-Flsun-Speeder-Pad#132] [Q&A]
I have a usb webcam attached ti speederpad. It works perfectly since some months ago when it become black

To get it work I have to log in via SSH and restart crownest service with 
"sudo systemctl restart crowsnest.service"
after that it works but only until i turn off the printer. Next time I'll have to log in ssh and restart again...

any suggestion? thanks

**Comment by Lumbi77** [comment]
I have the same Problem.

**Comment by dosoft** [comment]
Any thoughts on this? I'm dealing with the same issue, and it's pretty annoying having to restart this service every time I restart the printer

**Comment by Guilouz** [comment]
Remove and reinstall crowsnest.

**Comment by dosoft** [comment]
I've just reinstalled Crowsnest, but unfortunately, it didn't resolve the issue.

I've gone ahead and enabled debug logging, and this is what I've discovered at the end of the Crowsnest log:

```
[10/23/23 02:11:39] crowsnest: DEBUG: ustreamer [cam 1]: -- INFO  [18.352      main] -- Enabling HTTP file server: /home/pi/crowsnest/ustreamer-www
[10/23/23 02:11:39] crowsnest: DEBUG: ustreamer [cam 1]: [warn] getaddrinfo: address family for nodename not supported
[10/23/23 02:11:39] crowsnest: DEBUG: ustreamer [cam 1]: -- ERROR [18.352      main] -- Can't bind HTTP on [127.0.0.1]:8080: Success
[10/23/23 02:11:39] crowsnest: ERROR: Start of ustreamer [cam 1] failed!
```

Interestingly, when the service is restarted, everything appears to be working fine.

**Comment by dosoft** [comment]
Actually, this is an issue with the crowsnest - https://github.com/mainsail-crew/crowsnest/issues/117

I've added 10 secs timeout and with this workaround it works.

**Reply by AbitNutzT00** [reply]
where do you add the timeout? I'm having the same problem.
thanks

**Reply by dosoft** [reply]
By the way, I encountered this problem again, but this time it was related to a missing 'ustream' package. Please check the 'crowsnest' log; you might be experiencing the same issue. Reinstalling 'crowsnest' resolved it for me.

**Comment by dahimo** [comment]
> I have a usb webcam attached ti speederpad. It works perfectly since some months ago when it become black
> 
> To get it work I have to log in via SSH and restart crownest service with "sudo systemctl restart crowsnest.service" after that it works but only until i turn off the printer. Next time I'll have to log in ssh and restart again...
> 
> any suggestion? thanks

**Workaround on my V400**
Create a shell script that waits 30 seconds and then restarts the crowsnest service automatically:
```
#!/bin/bash

# Wait xx seconds
sleep 30

# Restart crowsnest
sudo systemctl restart crowsnest.service

```
Test the script if it works for you. Then add the following line to the file /etc/rc.local:
`<absolute path to your script>/<name of your script>.sh`
for example:
`/home/pi/scripts/restartCrowsnest.sh`
